backend/src/main/java/com/beaplab/BeaplabEngine/controller/WatchController.java

```diff
@@ -0,0 +1,336 @@
+                case "fitbit":
+                    viewName = "fitbit";
+                    break;
+                default:
+                    rollbar.error("Incorrect watch type selected: " + watchType);
+                    throw new Exception("Incorrect watch type selected: " + watchType);
+            }
+
+            ModelAndView model = new ModelAndView(viewName + "FileUpload");
+            return model;
+    }
+
+
+
+    // upload
+    @RequestMapping(value = "/{watchType}/upload", method = RequestMethod.POST)
+    @ApiOperation(value = "Zip file upload", notes = "Uploading a zip file containing Apple Watch raw data and persisting it to database")
+    @ApiResponses(value = {
+            @ApiResponse(code = 415, message = "{success: false, message: Invalid file type}"),
+            @ApiResponse(code = 200, message = "{success: true}, raw_data_id: XXXX}"),
+            @ApiResponse(code = 406, message = "{success: false, message: One zip file is acceptable}"),
+            @ApiResponse(code = 500, message = "{success: false, message: XYZ}")
+    })
+    @ApiImplicitParams({
+            @ApiImplicitParam(name = "file", value = "the zipped file to be uploaded as multipart/form-data, " +
+                    "being processData: false, contentType: false in the ajax request.\n" +
+                    "A sample ajax request is as follows:\n\n" +
+                    "--------begin example-------\n" +
+                    "var formData = new FormData();\n" +
+                    // not sure how to do this with general controller...
+                    "formData.append('fname', 'applewatch.zip');\n" +
+                    "formData.append('data', blob);\n" +
+                    "$.ajax({\n" +
+                    "type: 'POST',\n" +
+                                                // not sure here either
+                    "                            url: '/rest/beapengine/applewatch/upload',\n" +
+                    "                            data: formData,\n" +
+                    "                            processData: false,\n" +
+                    "                            contentType: false,\n" +
+                    "                            success: function (data, status, xhr) {\n" +
+                    "                                console.log('status: ' + status + ', data: ' + data);\n" +
+                    "                            },\n" +
+                    "                            error: function (jqXhr, textStatus, errorMessage) {\n" +
+                    "                                var result = JSON.parse(jqXhr.responseText);\n" +
+                    "                                console.log(\"error: \" + result)\n" +
+                    "                            }\n" +
+                    "                        });\n" +
+                    "--------end example-------\n",
+                    type = "POST",
+                    required = true, paramType = "blob")
```

Comment:
I have no idea what this is, or how to generalize it (see line 103 and 98)

Reply:
`@ApiImplicitParams` Comes from swagger and is just purely just for documentation purposes. Swagger is the library we have the automatically eveyrhting about our api routes by reading annotations such ass ApiImplicitParams.

So no need to necesarrily generalize but you can probably clean up the message here more

Probably good to just leave it as 

```
@ApiImplicitParams({
    @ApiImplicitParam(
        name = "file",
        value = "the zipped file to be uploaded as multipart/form-data",
        type = "POST",
        required = true,
        paramType = "blob"
    )
})
```